src/dialect/mod.rs

```diff
@@ -675,7 +610,20 @@ mod tests {
         struct WrappedDialect(MySqlDialect);
 
         impl Dialect for WrappedDialect {
-            fn dialect(&self) -> std::any::TypeId {
+            fn settings(&self) -> DialectSettings {
+                let s = self.0.settings();
+                DialectSettings {
```

Comment:
this is slightly ugly because `WrappedDialect` only implemented a few of the methods on the dialect.

But this actually demonstrates one of the advtanges of the approach in this PR:

If this test had been written with the new behaviour, this method would become

```rs
fn settings(&self) -> DialectSettings {
  let mut s = self.0.settings();
  s.custom_thing = true;
  s
}
```

And it would stay up-to-date even as new setting were added. Unlike the current behaviour.

I can update this test if you like?